python/friesian/src/bigdl/friesian/feature/table.py

```diff
@@ -34,29 +36,47 @@
     lit, rank, monotonically_increasing_id, row_number, desc
 from pyspark.sql.types import ArrayType, DataType, StructType, StringType, StructField
 
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+
+if TYPE_CHECKING:
+    from pandas.core.frame import DataFrame as PandasDataFrame
+    from pyspark.sql.column import Column
+    from pyspark.sql import Row
+    from pyspark.sql.dataframe import DataFrame as SparkDataFrame
+    from pyspark.sql.types import StructType
+
+
 JAVA_INT_MIN = -2147483648
 JAVA_INT_MAX = 2147483647
 
 
 class Table:
-    def __init__(self, df):
+    def __init__(self, df: "SparkDataFrame") -> None:
         self.df = df
         self.__column_names = self.df.schema.names
 
     @property
-    def schema(self):
+    def schema(self) -> "StructType":
         return self.df.schema
 
     @staticmethod
-    def _read_parquet(paths):
+    def _read_parquet(paths: Union[List[str], str]) -> "SparkDataFrame":
         if not isinstance(paths, list):
             paths = [paths]
         spark = OrcaContext.get_spark_session()
         df = spark.read.parquet(*paths)
         return df
 
     @staticmethod
-    def _read_json(paths, cols):
+    def _read_json(paths: str, cols: Union[List, str]):
```

Comment:
-> "SparkDataFrame" ?

Reply:
>  "SparkDataFrame" ?

For there are two kinds of DataFrame in this module: `pyspark.sql.dataframe.DataFrame and pandas.core.frame.DataFrame`. To avoid ambiguity of type `DataFrame`, I rename pyspark.sql.dataframe.DataFrame t o `SparkDataFrame` like:
```
from pyspark.sql.dataframe import DataFrame as SparkDataFrame
```
And so do PandasDataFrame.
Or should we just assign full name to avoid confusion like:
```
-> pyspark.sql.dataframe.DataFrame
```

Reply:
> -> "SparkDataFrame" ?

For there are two kinds of DataFrame in this module:` pyspark.sql.dataframe.DataFrame `and `pandas.core.frame.DataFrame`. To avoid ambiguity of type `DataFrame`, I rename `pyspark.sql.dataframe.DataFrame t` o SparkDataFrame like:
```
from pyspark.sql.dataframe import DataFrame as SparkDataFrame
```
And so do PandasDataFrame.
Or should we just assign full name to avoid confusion like:
```
-> pyspark.sql.dataframe.DataFrame
```

Reply:
`_read_json` should return SparkDataFrame, just like `_read_csv`, yes?

Reply:
> `_read_json` should return SparkDataFrame, just like `_read_csv`, yes?

yes, have fixed but not pushed

Reply:
> > `_read_json` should return SparkDataFrame, just like `_read_csv`, yes?
> 
> yes, have been fixed but not pushed yet